gematria/datasets/convert_bhive_to_llvm_exegesis_input.cc

```diff
@@ -36,10 +38,65 @@ constexpr std::string_view kMemDefPrefix = "# LLVM-EXEGESIS-MEM-DEF ";
 constexpr std::string_view kMemMapPrefix = "# LLVM-EXEGESIS-MEM-MAP ";
 constexpr std::string_view kMemNamePrefix = "MEM";
 
+enum class AnnotatorType { kExegesis, kFast };
+
+constexpr std::array<std::pair<AnnotatorType, std::string_view>, 2>
+    kAnnotatorTypeNames = {{{AnnotatorType::kExegesis, "exegesis"},
+                            {AnnotatorType::kFast, "fast"}}};
+
+bool AbslParseFlag(absl::string_view text, AnnotatorType* type,
+                   std::string* error) {
+  for (const auto& [annotator_type, type_string] : kAnnotatorTypeNames) {
+    if (text == type_string) {
+      *type = annotator_type;
+      return true;
+    }
+  }
+
+  *error = "unknown annotator type";
+  return false;
+}
+
+std::string AbslUnparseFlag(AnnotatorType type) {
+  for (const auto& [annotator_type, type_string] : kAnnotatorTypeNames) {
+    if (annotator_type == type) return std::string(type_string);
+  }
+
+  __builtin_unreachable();
+}
+
 ABSL_FLAG(std::string, bhive_csv, "", "Filename of the input BHive CSV file");
 ABSL_FLAG(
     std::string, output_dir, "",
     "Directory containing output files that can be executed by llvm-exegesis");
+ABSL_FLAG(AnnotatorType, annotator_implementation, AnnotatorType::kFast,
+          "The annotator implementation to use.");
+
+std::optional<gematria::AccessedAddrs> GetAccessedAddrs(
+    absl::Span<const uint8_t> basic_block,
+    gematria::ExegesisAnnotator* exegesis_annotator) {
+  const AnnotatorType annotator_implementation =
+      absl::GetFlag(FLAGS_annotator_implementation);
+  switch (annotator_implementation) {
+    case AnnotatorType::kFast: {
+      // This will only get the first segfault address.
+      auto addrs_fast = gematria::FindAccessedAddrs(basic_block);
```

Comment:
See gematria/llvm/llvm_to_absl.h, maybe add these backward converters there? same for line 93.

Reply:
`gematria::FindAccessedAddrs` already returns a `absl::StatusOr`. We could switch line 93, but I figured since we're just checking the error within the case and returning the actual value, it didn't make a lot of sense.

Reply:
No, I mean add a  `absl::StatusOr` -> `std::optional` converter and then reuse it here and below

Reply:
I've added a `absl::StatusOr` to `std::optional` converter in `llvm_to_absl.h`. I've used `llvm::expectedToOptional` to convert the expected to an optional value.

Reply:
Since it is an error, should we convert to `llvm::Expected`?

Reply:
I originally structured it this way given that the handler had additional useful information to report (in the calling function) that didn't make sense to send through here. Errors were originally reported by printing them out, but with the refactoring to the converters, that went away.

I guess we should probably just be returning a `absl::StatusOr` and converting the expected into that. Not sure why I didn't think of that in the first place.

Reply:
Fixed. Sorry for the back and forth here/misunderstanding on my end.